rct229/rulesets/ashrae9012019/section11/section11rule12.py

```diff
@@ -0,0 +1,150 @@
+                "schedules_p": find_all("$.schedules[*]", rmd_p),
+                "swh_uses_associated_with_each_building_segment_p": swh_uses_associated_with_each_building_segment_p,
+            }
+
+        class BuildingRule(PartialRuleDefinition):
+            def __init__(self):
+                super(Section11Rule12.RMDRule.BuildingRule, self).__init__(
+                    rmds_used=produce_ruleset_model_description(
+                        USER=False, BASELINE_0=True, PROPOSED=True
+                    ),
+                    manual_check_required_msg=APPLICABILITY_MSG,
+                )
+
+            def is_applicable(self, context, data=None):
+                building_b = context.BASELINE_0
+                building_p = context.PROPOSED
+                schedules_b = data["schedules_b"]
+                is_leap_year_b = data["is_leap_year_b"]
+
+                service_water_heating_uses_p = find_all(
+                    "$.building_segments[*].zones[*].spaces[*].service_water_heating_uses[*]",
+                    building_p,
+                )
+                building_open_schedule_p = find_one(
+                    "$.building_open_schedule", building_p
+                )
+
+                bldg_open_sch_id_b = getattr_(
+                    building_b, "buildings", "building_open_schedule"
+                )
+
+                bldg_open_sch_b = next(
+                    (
+                        schedule_b["hourly_values"]
+                        for schedule_b in schedules_b
+                        if schedule_b["id"] == bldg_open_sch_id_b
+                    ),
+                    [],
+                )
+
+                hours_this_year = (
+                    LeapYear.LEAP_YEAR_HOURS
+                    if is_leap_year_b
+                    else LeapYear.REGULAR_YEAR_HOURS
+                )
+
+                return (
+                    service_water_heating_uses_p
+                    and building_open_schedule_p is not None
+                    and sum(bldg_open_sch_b) == hours_this_year
+                )
+
+            def get_calc_vals(self, context, data=None):
+                swh_uses_associated_with_each_building_segment_p = data[
+                    "swh_uses_associated_with_each_building_segment_p"
+                ]
+
+                return {
+                    "swh_uses_associated_with_each_building_segment_p": swh_uses_associated_with_each_building_segment_p
+                }
+
+            def applicability_check(self, context, calc_vals, data):
+                building_p = context.PROPOSED
+
+                swh_uses_associated_with_each_building_segment_p = calc_vals[
+                    "swh_uses_associated_with_each_building_segment_p"
+                ]
+
+                service_water_heating_uses_list_p = []
+                for bldg_segment_id in swh_uses_associated_with_each_building_segment_p:
+                    for (
+                        served_by_distribution_system_id
```

Comment:
I am very confused by this `served_by_distribution_system_id` while the parameter name is `swh_uses_associated_with_each_building_segment_p`

Reply:
Good point. Addressed.